File: Models/UserModel.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HospitalManagement.Models
{
    public class UserModel
    {
        public long? id { get; set; }
        public string name { get; set; }
        public string username { get; set; }
        public string password { get; set; }
        public string email { get; set; }
        public string phoneNumber { get; set; }
        public string gender { get; set; }
        public string role { get; set; }
        public IEnumerable<string> roles { get; set; }
        public int? age { get; set; }
        public string country_name { get; set; }
        public string country_short_name { get; set; }
        public int? country_phone_code { get; set; }
        public string state_name { get; set; }
        public string city_name { get; set; }
        public string bloodGroup { get; set; }
        public string bmdc_certifcate { get; set; }
        public IFormFile profilePic { get; set; }
    }
}

File: ClientApp/src/app/models/user.model.ts

import { Inject } from "@angular/core";
import { DoctorAppointment } from "./doctor-appointment.model";
import { Language } from "./langauge.model";
import { Schedule } from "./schedule.model";
import { Speciality } from "./speciality.model";

export class User{

  public baseUrl:string;


  public id:number;
  public name:string;
  public username:string;
  public email:string;
  public age:number;
  public gender:string;
  public password:string;
  public new_password: string;
  public roles:string[] = [];
  public role:string;
  public profile_pic: string | ArrayBuffer;
  public phoneNumber: string;
  public isActive: boolean;
  public city_name: string;
  public country_name: string;
  public country_phone_code: number;
  public country_short_name: string;
  public state_name: string;
  public bloodGroup: string;
  public created_date:Date;
  public address: string;


  //appointment info
  appointment: DoctorAppointment;



  // doctor info
  public bmdc_certifcate: string;
  public approved: boolean = false;
  public doctor_title: string;
  public degree_title: string;
  public biography: string;
  public specialities: Speciality[] = [];
  public experience: number;
  public types_of: string;
  public languages: Language[] = [];
  public schedules: Schedule[] = [];
  public new_patient_visiting_price: number;
  public old_patient_visiting_price: number

  get profielPicById(){

     var image_url = this.baseUrl + 'api/usermanager/GetProfilePic?id=' + this.id.toString();
     return  `background-image: url(${image_url});`;

    }
}
